tidb_vector/utils.py

```diff
@@ -1,34 +1,27 @@
+import typing
+
 import numpy as np
 
 
-def encode_vector(value, dim=None):
+def encode_vector(value: typing.Union[np.ndarray, typing.List[float]], dim=None):
     if value is None:
         return value
 
+    if dim is not None and len(value) != dim:
+        raise ValueError(f"expected {dim} dimensions, but got {len(value)}")
+
     if isinstance(value, np.ndarray):
         if value.ndim != 1:
             raise ValueError("expected ndim to be 1")
+        return f"[{','.join(map(str, value))}]"
 
-        if not np.issubdtype(value.dtype, np.integer) and not np.issubdtype(
-            value.dtype, np.floating
-        ):
-            raise ValueError("dtype must be numeric")
+    return str(value)
 
-        value = value.tolist()
-
-    if dim is not None and len(value) != dim:
-        raise ValueError("expected %d dimensions, not %d" % (dim, len(value)))
 
-    return "[" + ",".join([str(float(v)) for v in value]) + "]"
-
-
-def decode_vector(value):
-    if value is None or isinstance(value, np.ndarray):
+def decode_vector(value: str) -> np.ndarray:
+    if value is None:
         return value
 
-    if isinstance(value, bytes):
```

Comment:
This is necessary when using mysqlclient

Reply:
Thanks for pointing out! It might be a stale behavior. Since 2023 Jul, VECTOR type was returned as non-binary LONGTEXT (see https://github.com/tidbcloud/tidb-cse/pull/1066/files), so this could be avoided. I also verified that by using `mysql+mysqlconnector` tests_sqlalchemy works fine as well.